src/slicer.h

```diff
@@ -109,12 +116,350 @@ class SlicerLayer
 
     /*!
      * Try to close up polylines into polygons while they have large gaps in them.
-     * 
+     *
      * Clears all open polylines which are used up in the process
-     * 
+     *
      * \param[in,out] open_polylines The polylines which are stiched, but couldn't be closed into a loop yet
      */
     void stitch_extensive(Polygons& open_polylines);
+
+private:
+    /*!
+     * \brief This class represents the location of an end point of a
+     * polyline in a polyline vector.
+     *
+     * The location records the index in the polyline vector and
+     * whether this is the vertex at the start of the polyline or the
+     * vertex at the end.
+     */
+    class Terminus
+    {
+    public:
+        /*! A representation of Terminus that can be used as an array index. */
```

Comment:
Ok, so there is a particular array into which this is indexing: `TerminusTrackingMap::m_terminus_old_to_cur_map` and `TerminusTrackingMap::m_terminus_cur_to_old_map`

Saying that the Index can be used as "an array index" is rather non-specific. Since you expose the `Index` functionality specially to be used with a TerminusTrackingMap, it would be nice to document that here.